### PR TITLE
Refine Python docs

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

#1600 Modify Python docs (some typos, formats, python versions).

### CODE REVIEW

File: docs/docs/release-download.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 
 
-These are built BigDL packages including dependency and python files. You can download these packages instead of building them by yourself. This is useful when you want to do something like run some examples or develop python code.
+These are built BigDL packages including dependency and Python files. You can download these packages instead of building them by yourself. This is useful when you want to do something like run some examples or develop Python code.
 
+NOTE: __Python 2.7__ and __Python 3.5__ are recommended.
```
Comment (jason-dai) [inline docs/docs/release-download.md]: Only python 2.7 and 3.5 are currently supported.

Comment (hkvision) [reply docs/docs/release-download.md]: OK~

File: docs/docs/PythonUserGuide/install-without-pip.md [hunk 2]
```diff
@@ -5,12 +5,12 @@
 2. You can download the BigDL release and nightly build from the [Release Page](../release-download.md)
```
Comment (jason-dai) [inline docs/docs/PythonUserGuide/install-without-pip.md]: Add a note that only python 2.7 and 3.5 are currently supported.

Comment (hkvision) [reply docs/docs/PythonUserGuide/install-without-pip.md]: Added and highlighted.

File: docs/docs/PythonUserGuide/install-from-pip.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ## **NOTES**
 
-- Pip install support __mac__ and __linux__ platform but only __Spark1.6.x__ for now.
-- Pip install only support run in __local__. Might support cluster mode in the future.
-- We've tested this package with __python 2.7__ and __python 3.5__
+- Pip install supports __Mac__ and __Linux__ platforms but only __Spark1.6.x__ for now.
+- Pip install only supports __local__ mode. Might support cluster mode in the future.
```
Comment (jason-dai) [inline docs/docs/PythonUserGuide/install-from-pip.md]: Add a link to "install without pip" for people to use it in cluster mode.

Comment (hkvision) [reply docs/docs/PythonUserGuide/install-from-pip.md]: Updated.

**Comment by jason-dai** [comment]
LGTM

**Pull request merged**